util/ttl_map.go

```diff
@@ -0,0 +1,56 @@
+package util
+
+import (
+	"sync"
+	"time"
+)
+
+type item struct {
+	value     string
+	createdAt int64
+}
+
+type TTLMap struct {
+	m map[string]*item
+	l sync.Mutex
+}
+
+func NewTTLMap(maxTTL int64, checkFrequency int64) (m *TTLMap) {
+	m = &TTLMap{m: make(map[string]*item)}
+	go func() {
+		for now := range time.Tick(time.Second * time.Duration(checkFrequency)) {
+			m.l.Lock()
+			for k, v := range m.m {
+				if now.Unix()-v.createdAt > maxTTL {
+					delete(m.m, k)
+				}
+			}
+			m.l.Unlock()
```

Comment:
NBD, but I think idiomatic Go defers the unlock right after the lock to prevent leaked locks if, e.g., code maintenance introduces early returns. I.e.

```golang
m.l.Lock()
defer m.l.Unlock()
```